wave_lang/kernel/wave/templates/extend_attention.py

```diff
@@ -260,23 +208,26 @@ def first_loop(
             q_reg = tkw.read(
                 q,
                 elements_per_thread=LOAD_ELEMS_PER_THREAD_QK,
-                mapping=q_mapping,
+                source=(n_q + EXT_IDX, h, d_q),
```

Comment:
Do the position of iterators matter? In the mask mapping `N_Q` was the second dimension but here it's the first

Reply:
yes, the order of the iterators should match the shape. So here the shape of the source would be (N_Q, H, D_Q).